analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotCheckZeroSizeCollection.cs

```diff
@@ -33,14 +33,29 @@ namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
-    public sealed class DoNotCheckZeroSizeCollection : DoNotCheckZeroSizeCollectionBase<SyntaxKind, BinaryExpressionSyntax, ExpressionSyntax>
+    public sealed class DoNotCheckZeroSizeCollection : DoNotCheckZeroSizeCollectionBase<ExpressionSyntax, SyntaxKind>
     {
         protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
-        protected override SyntaxKind GreaterThanOrEqualExpression => SyntaxKind.GreaterThanOrEqualExpression;
-        protected override SyntaxKind LessThanOrEqualExpression => SyntaxKind.LessThanOrEqualExpression;
-        protected override SyntaxKind GreaterThanExpression => SyntaxKind.GreaterThanExpression;
-        protected override SyntaxKind LessThanExpression => SyntaxKind.LessThanExpression;
-        protected override string IEnumerableTString { get; } = "IEnumerable<T>";
+        protected override string IEnumerableTString => "IEnumerable<T>";
+        protected override ExpressionSyntax Left(SyntaxNode binary) => ((BinaryExpressionSyntax)binary).Left;
+        protected override ExpressionSyntax Right(SyntaxNode binary) => ((BinaryExpressionSyntax)binary).Right;
+        protected override ExpressionSyntax Trim(ExpressionSyntax expression)
+        {
+            while (expression.RemoveParentheses() is ConditionalAccessExpressionSyntax conditionalAccess)
+            {
+                expression = conditionalAccess.WhenNotNull;
+            }
+            return (expression is InvocationExpressionSyntax invocation
+                ? invocation.Expression
+                : expression).RemoveParentheses();
```

Comment:
let's make this single line, the parentheses are confusing to read on multiline statement

Reply:
Do we want RemoveParentheses on the LanguageFacade? I think it used a lot.